oxenss/storage/database.cpp

```diff
@@ -287,6 +287,30 @@ ALTER TABLE messages ADD COLUMN namespace INTEGER NOT NULL DEFAULT 0;
             )");
         }
 
+        if (!db.tableExists("revoked_subkeys")) {
+            log::info(logcat, "Upgrading database schema");
+            db.exec(R"(
+CREATE TABLE revoked_subkeys (
+    id INTEGER,
+    owner INTEGER REFERENCES owners(id) ON DELETE CASCADE,
+    subkey TEXT NOT NULL,
+    timestamp INTEGER NOT NULL DEFAULT CURRENT_TIMESTAMP,
+
+    UNIQUE(subkey)
```

Comment:
```suggestion
    PRIMARY KEY(owner, subkey)
```
Revoked subkeys are unique per owner, but not necessarily globally.  Having the first indexed column on `owner` is also important for the queries in the trigger (without it it would have to do a full table scan every time to find matching rows).